Cql/Cql.Compiler/CqlOperatorsBinding.Bind.cs

```diff
@@ -0,0 +1,121 @@
+            {
+                for (int argIndexForGenericMethod = 0;
+                     argIndexForGenericMethod < Math.Min(args.Length, 2);
+                     argIndexForGenericMethod++) // Try to get generic type from argument up to the second one
+                {
+                    MethodInfo bindMethod;
+                    if (method is { IsGenericMethodDefinition: true })
+                    {
+                        if (!args[argIndexForGenericMethod].Type.IsGenericType)
+                            continue; // Not a generic argument, try again
+
+                        var genericType = args[argIndexForGenericMethod].Type.GetGenericArguments()[0];
+                        bindMethod = method.MakeGenericMethod(genericType);
+                    }
+                    else
+                    {
+                        bindMethod = method;
+                    }
+
+                    ParameterInfo[] methodParameters = bindMethod.GetParameters();
+                    if (methodParameters.Length != args.Length)
+                        continue;
+
+                    bool ConvertBindArguments()
+                    {
+                        for (int i = 0; i < args.Length; i++)
+                        {
+                            if (!TryConvert(args[i].Type, methodParameters[i].ParameterType, args[i], out bindArgs[i]!))
+                                return false;
+                        }
+                        return true;
+                    }
+
+                    if (!ConvertBindArguments())
+                        continue;
+
+                    var call = Expression.Call(CqlContextExpressions.Operators_PropertyExpression, bindMethod, bindArgs);
+                    return call;
+                }
+            }
+
+            if (i == 0 && args[^1] is ConstantExpression { Value: null })
+            {
+                // Handles precision cases where the last argument might be supplied or not
+                args = args[..^1];
+                continue;
+            }
+
+            break;
+        }
+        return null;
+    }
+
+
+    private MethodCallExpression BindToMethodConvertArgs(
+        string methodName,
+        params Expression[] arguments)
+    {
+        return BindToMethodConvertArgsOrNull(methodName, arguments)
+            ?? throw NoMethodFoundException();
+
+        ArgumentException NoMethodFoundException()
+        {
+            var types = string.Join(", ", arguments.Select(e => e.Type.Name));
+            return new ArgumentException($"No suitable method found {methodName}({types}).", nameof(methodName));
+        }
+    }
+
+    private static MethodCallExpression BindToMethod(
+        string methodName,
+        params Expression[] arguments) =>
+        Expression.Call(CqlContextExpressions.Operators_PropertyExpression, methodName, null, arguments);
+
+    private static MethodCallExpression BindToMethod(
+        string methodName,
+        Type[] typeArguments,
```

Comment:
For readability, can we change the name `typeArguments` to `generic(Type)Arguments`?

Reply:
I think I will also rename `BindToMethod` to `BindToGenericMethod`